File: teoricos/tt4/OrdenacaoInternaSelecao/Codigo/java/Dados_TExec/tempoMedio.java
class tempoMedio {
    
    public static char replaceVirgula(char c)
    {
        return (c == ',' ? '.' : c);
    }

    public static String replaceString(String in)
    {
        String out = "";

        for(int i = 0; i < in.length(); i++)
        {
            out += replaceVirgula(in.charAt(i));
        }

        return out;
    }

    public static void main (String[] args)
    {
        Arq file = new Arq();
        
        file.openRead("decrescente10000.txt");

        int count = 0;
        double num = 0,
               media = 0;

        while(file.hasNext())
        {
            num = Double.parseDouble(replaceString(file.readLine()));
            System.out.println(num);
            count++;
            //System.out.println("Media = "+ media);
            media += num;
        }
         
        //System.out.println("Count = "+ count);
        media = media/count;
        System.out.println("Media = "+ media);

        file.close();

    }
}

File: teoricos/tt4/OrdenacaoInternaSelecao/Codigo/c/tempoMedio.c
#include <stdio.h>

int main()
{   
    // Ler aquivo
    FILE *arq = fopen("decrescente10000.txt", "r");


    double num = 0;
    double media = 0;
    int count = 0;

    fscanf(arq, "%lf", &num);
    count++;
    while(!feof(arq))
    { 
        media += num; 

        fscanf(arq, "%lf", &num);
        count++;
    }
    count--; // tira a leitura do fim do arquivo 
        
    media = media/count;

    fclose(arq);

    printf("\n%lf\n", media);
    return 0;
}
